src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java

```diff
@@ -57,69 +63,19 @@ public static void main(String [] args) throws Exception {
-                                                    "Jello Biafra And The Guantanamo School Of "
-                                                    + "Medicine",
-                                                    "Three Strikes",
-                                                    "Punk"),
-                                           new Song(7L,
-                                                    "Licensed to Ill",
-                                                    "The Beastie Boys",
-                                                    "Fight For Your Right",
-                                                    "Hip Hop"),
-                                           new Song(8L,
-                                                    "De La Soul Is Dead",
-                                                    "De La Soul",
-                                                    "Oodles Of O's",
-                                                    "Hip Hop"),
-                                           new Song(9L,
-                                                    "Straight Outta Compton",
-                                                    "N.W.A",
-                                                    "Gangsta Gangsta",
-                                                    "Hip Hop"),
-                                           new Song(10L,
-                                                    "Fear Of A Black Planet",
-                                                    "Public Enemy",
-                                                    "911 Is A Joke",
-                                                    "Hip Hop"),
-                                           new Song(11L,
-                                                    "Curtain Call - The Hits",
-                                                    "Eminem",
-                                                    "Fack",
-                                                    "Hip Hop"),
-                                           new Song(12L,
-                                                    "The Calling",
-                                                    "Hilltop Hoods",
-                                                    "The Calling",
-                                                    "Hip Hop")
 
-    );
+    // Read comma-delimited file of songs into Array
+    List<Song> songs = new ArrayList<>();
+    final String SONGFILENAME= "song_source.csv";
+    InputStream inputStream = KafkaMusicExample.class.getClassLoader().getResourceAsStream(SONGFILENAME);
+    InputStreamReader streamReader = new InputStreamReader(inputStream, UTF_8);
+    BufferedReader br = new BufferedReader(streamReader);
+    for (String line; (line = br.readLine()) != null;) {
+      String[] values = line.split(",");
+      Song newSong = new Song(Long.parseLong(values[0]),values[1],values[2],values[3],values[4]);
+      songs.add(newSong);
+    }
+    br.close();
```

Comment:
We should use a try-with-resources to ensure it gets closed.

Reply:
I reworked the code...how is rev 2?